core/src/main/java/org/infinispan/metrics/impl/InfinispanMetricsRegistry.java

```diff
@@ -133,4 +123,16 @@ public void unregister(MetricID metricId) {
                     metricId, System.identityHashCode(registry), registry.getMetrics().size());
       }
    }
+
+   private static final Pattern SNAKE_CASE_PATTERN = Pattern.compile("(?<=[a-z])[A-Z]");
+
+   private static String decamelize(String s) {
+      Matcher m = SNAKE_CASE_PATTERN.matcher(s);
+      StringBuffer sb = new StringBuffer();
```

Comment:
StringBuilder !

Reply:
I've just updated that. The problem is the Matcher method that takes a stringbuilder instead of stringbuffer is jdk 9. So I reimplemented that without Pattern/Matcher as it's not worth the hassle